### ISSUE TITLE
Apps are duplicated

### ISSUE THREAD

**Issue by adamdriscoll** [ironmansoftware/powershell-universal#3102]
### Version

4.2.12

### Severity

High

### Steps to Reproduce

This is a regression and was historically hard to reproduce but apps can get duplicated. It seems to happen when a user does one of the following things: 
- Enable git edit mode
- Delete scripts or apps

Restarting the service fixes the issue.

### Expected behavior

```console
No duplication.
```

### Actual behavior

```console
Duplication.
```

**Comment by tylermiranda** [comment]
Hey Adam, we are seeing this too.  Just to add, we are see it when one of the .universal configuration files is modified and then a git sync occurs.  Here is an example of my exact workflow in case it may be of help in narrowing down the issue.
1. Modify environments.ps1, adding a new environment.
2. Commit and push the change to the remote github repo.
3. Trigger a git sync through the PSU management API.

This doesn't always cause it to happen but its not hard to reproduce.  We've also noticed that when it works and doesn't duplicate the apps it looks like PSU is reloading for a moment. Meaning when we refresh the apps page, we will briefly seeing the loading screen that you get when the node is starting up.  Lastly, we only started having this happen when we brought up a second psu node sharing the same SQL database and git repo.

We are running 4.2.12 under IIS but we also saw it on 4.2.3.

**Comment by adamdriscoll** [comment]
That's super helpful. Thank you!

**Comment by adamdriscoll** [comment]
@tylermiranda - I'm still having a hard time reproducing this. Can you please try enabling debug system logs? 

https://support.ironmansoftware.com/portal/en/kb/articles/kb0061-enabling-debug-system-logs

Any info you can share on your Repository folder would be good too. If it's something you can share, feel free to add the details to an email to support[at]ironmansoftware.com

**Comment by tylermiranda** [comment]
@adamdriscoll I've enabled debug logging as suggested from the link.  I'm not sure it's working though.  Its not honoring SystemLogPath location and instead creating them at the root path (C:\PSU).  The log files it is creating are all informational. 
![image](https://github.com/ironmansoftware/issues/assets/254684/041d072c-a720-4ad2-a3d8-ed3ec78a9662)


This is running under IIS by the way.  I stopped and started the app pools after making the change in the appsettings.json file.

**Comment by tylermiranda** [comment]
Ok, I noticed your example didn't have the SystemLogPath in the json file. I removed that entry and am now seeing debug info.  I'll reproduce and email you the logs

**Comment by tylermiranda** [comment]
Interestingly enough, since I have enabled debug logging I havent been able to reproduce.  After several tries manually, I ended up writing a script that touches the environments.ps1 file, syncs the change to git, hits the management api to get the number of dashboards and loops if the number is correct.  It runs once every 60s and is currently on try 66 without reproducing the issue. I dont know...

**Comment by tylermiranda** [comment]
@adamdriscoll i sent an email to the address listed above with the log file attached. It was large so I had to use my secure send service.

**Comment by adamdriscoll** [comment]
Still trying to track this done. Moving into this milestone.

**Comment by adamdriscoll** [comment]
I nailed it down. I was able to reproduce this with about 5 PSU instances, a single database and git sync. I had 10 apps and was making changes locally and within github and it would eventually trigger. 

Ultimately, I discovered that apps were being restarted frequently on all nodes and it was cause all kinds of weird issues. This was triggered by changes in git sync. This change should also really help with server startup times.
